packages/peregrine/lib/talons/MiniCart/useMiniCart.js

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useMemo } from 'react';
+import { useQuery, useMutation } from '@apollo/react-hooks';
+
+import { useCartContext } from '../../context/cart';
+
+export const useMiniCart = props => {
+    const { queries, mutations } = props;
+    const { miniCartQuery } = queries;
+    const { removeItemMutation } = mutations;
+
+    const [{ cartId }] = useCartContext();
+
+    const {
+        data: miniCartData,
+        loading: miniCartLoading,
+        error: miniCartError
+    } = useQuery(miniCartQuery, {
+        fetchPolicy: 'cache-and-network',
+        variables: { cartId },
+        skip: !cartId
+    });
+
+    const [
+        removeItem,
+        { loading: removeItemLoading, called: removeItemCalled }
+    ] = useMutation(removeItemMutation);
+
+    const totalQuantity = useMemo(() => {
+        if (!miniCartLoading && miniCartData) {
+            return miniCartData.cart.total_quantity;
+        }
+    }, [miniCartData, miniCartLoading]);
+
+    const productList = useMemo(() => {
+        if (!miniCartLoading && miniCartData) {
+            return miniCartData.cart.items;
+        }
+    }, [miniCartData, miniCartLoading]);
+
+    const handleRemoveItem = useCallback(
+        async id => {
+            try {
+                const { error } = await removeItem({
+                    variables: {
+                        cartId,
+                        itemId: id
+                    }
+                });
+
+                if (error) {
+                    throw error;
```

Comment:
Check out how we're handling removal mutations in [Tommy's PR](https://github.com/magento/pwa-studio/blob/49cc9950a1a50f7c42cd3b8197dccde7e2a6e44b/packages/peregrine/lib/talons/CartPage/PriceAdjustments/useCouponCode.js#L55). We don't catch errors and we handle the messaging through deriving from the hook `error` object.

Reply:
One step further, I would say you only need to wrap mutations in a try/catch if you need block execution of the rest of the callback. I would say [this pattern](https://github.com/magento/pwa-studio/pull/2495/files#diff-f834782b75e39e20a9fc59a568a061d5R119-R143) is a better reference for doing that (just return), and no need to try/ catch otherwise, just rely on `error` in the `useMutation` response.

Reply:
Handled in a811662